src/exchange/moderators/truncate.py

```diff
@@ -47,15 +47,17 @@ def _update_system_prompt_token_count(self, exchange: Exchange) -> None:
 
         if not self.system_prompt_token_count or is_different_system_prompt:
             # calculate the system prompt tokens (includes functions etc...)
+            # we use a placeholder message with one token, which we subtract later
+            # this ensures compatible with providers that require a user message
             _system_token_exchange = exchange.replace(
-                messages=[],
+                messages=[Message.user("a")],
                 checkpoint_data=CheckpointData(),
                 moderator=PassiveModerator(),
                 model=self.model if self.model else exchange.model,
             )
             _system_token_exchange.generate()
             last_system_prompt_token_count = self.system_prompt_token_count
-            self.system_prompt_token_count = _system_token_exchange.checkpoint_data.total_token_count
+            self.system_prompt_token_count = _system_token_exchange.checkpoint_data.total_token_count - 1
```

Comment:
Just a heads up, this isn't quite right (and depends on model) since the addition of a user message adds some extra tokens to indicate the start/end of the message for the role (user/assistant). But this estimate is within ~5 tokens which is fine and is the best we can do without building out a token counting module for this

Reply:
yes that does make sense - i think that is alright though, because we're sort of redefining what "system_prompt_token_count" means by a bit - it is now both the system and the unavoidable overhead of sending the next message, which is still a reasonable thing to count towards the limit?